vscode/src/edit/provider.ts

```diff
@@ -42,23 +43,144 @@ import { isStreamedIntent } from './utils/edit-intent'
 interface EditProviderOptions extends EditManagerOptions {
     task: FixupTask
     controller: FixupController
+    cacheManager: EditCacheManager
+}
+
+/**
+ * Represents the possible states of a streaming edit session
+ */
+type StreamState = 'prefetching' | 'streaming' | 'completed'
+
+/**
+ * Data structure that captures an in-progress or completed streaming session
+ * for a given task. This allows both "prefetch" and "startEdit" to share
+ * partial text and avoid multiple LLM requests.
+ */
+export interface StreamSession {
+    /** Accumulates partial text as it comes in */
+    partialText: string
+    /** For streaming abort */
+    abortController: AbortController
+    /** In-flight streaming promise (so we won't start two streams for the same task) */
+    streamingPromise: Promise<void> | null
+    /** Multiplexer to broadcast partial content */
+    multiplexer: BotResponseMultiplexer
+    /** Current state of the streaming session */
+    state: StreamState
 }
 
-// Initiates a completion and responds to the result from the LLM. Implements
-// "tools" like directing the response into a specific file. Code is forwarded
-// to the FixupTask.
+/**
+ * We store the streaming session in a Map keyed by task ID so prefetchEdit
+ * and startEdit share the same stream if it's in progress.
+ */
 export class EditProvider {
     private insertionQueue: { response: string; isMessageInProgress: boolean }[] = []
     private insertionInProgress = false
-    private abortController: AbortController | null = null
 
     constructor(public config: EditProviderOptions) {}
 
+    /**
+     * Attempt to start streaming in "prefetch mode." If a stream is already
+     * in progress for this task, do nothing. The user has not clicked "apply"
+     * yet, so we do NOT call controller.startTask here.
+     */
+    public async prefetchEdit(): Promise<void> {
```

Comment:
```suggestion
    public prefetchEdit(): void {
```

We're not actually returning a promise here, is this intended? I see we `await` this function elsewhere.